server/sonar-db-dao/src/main/java/org/sonar/db/ideusage/IdeUsageDao.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.ideusage;
+
+import static org.sonar.db.DatabaseUtils.executeLargeInputsWithoutOutput;
+import static org.sonar.db.ideusage.IdeUsageMapper.getMapper;
+
+import java.util.List;
+import org.sonar.db.Dao;
+import org.sonar.db.DbSession;
+
+public class IdeUsageDao implements Dao {
+
+  public void deleteByOrganizationAndUser(DbSession dbSession, String organizationUuid, String userUuid) {
+    List<String> uuids = getMapper(dbSession).selectUuidsByOrganizationAndUser(organizationUuid, userUuid);
```

Comment:
@ritesh-ghiya-cs 

I don't get why you query all the data from DB into the memory and then remove it by uuids.

You can execute the delete query directly:

```
<delete id="deleteByOrganizationAndUser" parameterType="map" resultType="String">
    delete
    from cs_ide_usage iu
    where
    iu.user_uuid=#{userUuid,jdbcType=VARCHAR}
    and iu.organization_uuid=#{organizationUuid,jdbcType=VARCHAR}
  </delete>
```